packages/components/src/Notification/ToastNotification/subcomponents/ToastNotificationManager/ToastNotificationManager.tsx

```diff
@@ -0,0 +1,139 @@
+import React from "react"
+import { createRoot } from "react-dom/client"
+import { v4 } from "uuid"
+import {
+  AddToastNotification,
+  ClearToastNotifications,
+  RemoveToastNotification,
+  ToastNotification,
+  ToastNotificationWithOptionals,
+} from "../../types"
+
+import { ToastNotificationsListContainer } from "../ToastNotificationsListContainer"
+
+type ToastNotificationApi = {
+  addToastNotification: AddToastNotification
+  removeToastNotification: RemoveToastNotification
+  clearToastNotifications: ClearToastNotifications
+}
+
+/**
+ * Persistent reference to a "portal" for rendering the separate DOM tree
+ */
+let portal: HTMLDivElement | undefined
+
+/**
+ * Set default values for optional params
+ */
+const setDefaults = (
+  notificationMaybe: ToastNotificationWithOptionals
+): ToastNotification => {
+  const copy = { ...notificationMaybe }
+  if (typeof copy.id === "undefined") {
+    copy.id = v4()
+  }
+  return copy as ToastNotification
+}
+
+/**
+ * Create an instance of the toast notification manager
+ * Renders ToastNotifications in an independent React tree
+ * @returns { addToastNotification, removeToastNotification, clearToastNotifications }
+ */
+const createToastNotificationManager = (): ToastNotificationApi => {
+  let setNotifications:
+    | React.Dispatch<React.SetStateAction<ToastNotification[]>>
+    | undefined
+  if (portal === undefined && typeof window !== "undefined") {
+    // automation and testid appear to be consumed in other repos, we'll need to refactor before we remove so they can be added in
+    portal = document.createElement("div")
+    portal.setAttribute(
+      "data-automation-id",
```

Comment:
Are you replacing these with `data-testid`?

Reply:
my one concern is [this file](https://github.com/cultureamp/recognition-front-end/blob/d85ee88a09ad7c9f82d2e2f48d0674d6330ec504/apps/recognition-feed/src/templates/Admin/styles.scss#L42) that seems to use it as a selector. That said, doesn't feel like the best practice anyway and they could just change it to data-testid

Reply:
A good catch, but I'll add the guidance for the removal of `data-automation-id` in the migration docs so that everyone is aware. So we should remove this.

Reply:
Brill, will put the 🪓  to it!